department-of-veterans-affairs/caseflow
# Hun jer bah/appeals 3419

Resolves feature toggles needed for [APPEALS-3419](https://vajira.max.gov/browse/APPEALS-3419) release. 

### Description
Added feature toggles to key parts of APPEALS-3419 so that the workflow cannot be accessed until the toggle is turned on. 

### Acceptance Criteria
- [ ] Toggles cover the Split Appeal button
- [ ] Toggles cover the split appeal controller
- [ ] Toggles cover any other key area of 3419 workflow. 

### Testing Plan
1. Log into Caseflow local as a **SPLTAPPL** user to access the 3419 workflow. 
2. Go to Queue via the link
3. Click on the search bar at the top
4. Put in any veteran file number (5000000, 5000002, etc.) and click the veteran docket number to navigate to the case details page. 
5. Click on the **correct issues** link.
6. Open up the inspector of the web browser. You will need to use Google Chrome with the **React Developer Tools** extension ([located here if needed](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)) and open up the **Components** tab. Navigate down to the **AddIssuesPage** to view the page props. In the props window, click **featureToggles** to see the list of feature toggles. Confirm that the `split_appeal_workflow` toggle is there like in the picture below.
![image (4)](https://user-images.githubusercontent.com/99915461/193023203-e4012e19-10e8-48e9-ae8b-2c160f0c045d.png)
7. If needed, add 2 request issues to the Appeal and save. Return to the **AddIssues page** by clicking the **correct issues** link again. The **split appeal** button should now show. Again, navigate to the `split_appeal_workflow` feature toggle in the Components tab. Toggle the feature toggle to set it to **false** like in the picture above. The button should disappear.
8. Next, enable the button to go to the next page to view the next feature toggle. 
9. On the **Create new Appeal Stream** page, select a value for the dropdown box, and click a checkbox. The **continue** button should light up. Before clicking the button, navigate to the **Components** tab again in Chrome and go to **IntakeEditFrame**.
10. Click on **IntakeEditFrame**. Inside the props view, click **featureToggles** and find `split_appeal_workflow`. Click the toggle to set the value to **false**. The buttons on the bottom of the page should no longer be visible or clickable.c
11. If needed, continue through the split appeal workflow to the make sure that the rest of the workflow operates as it should.
12. Finally, confirm these places in the code that have the `split_appeal_workflow` toggles:

- [ ] Split Appeal Success/Fail banner
![image](https://user-images.githubusercontent.com/99915461/193025638-a55f91a2-0b54-443a-8382-5895d8e18f9b.png)

- [ ] Split Appeal Controller
![image](https://user-images.githubusercontent.com/99915461/193025703-12f15443-6dea-401e-8469-c165e5875cf4.png)

- [ ] Split appeal permission logic
![image](https://user-images.githubusercontent.com/99915461/193025896-3443056f-b87f-4435-afc9-bf5644794ac6.png)
